src/core/ledger/diffLedger.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+
+import {Ledger} from "./ledger";
+import type {LedgerEvent} from "./ledger";
+import diffBy from "lodash.differenceby";
+
+export opaque type LedgerDiff = $ReadOnlyArray<LedgerEvent>;
```

Comment:
Is there a reason we make the API asymmetrical? What if we did something like:

```js
export type LedgerDiff = {|
  +onlyInFirstLedger: $ReadOnlyArray<LedgerEvent>,
  +onlyInSecondLedger: $ReadOnlyArray<LedgerEvent>,
  // Events that had the same UUID but were not equal
  +uuidConflicts: $ReadOnlyArray<[LedgerEvent, LedgerEvent]>,
|}

Reply:
Yea that might be handy. Don't think we need the uuidConflicts as per my comment above, which would allow us to return a tuple for some smooth ergonomics:

```ts
export type LedgerDiff = [$ReadOnlyArray<LedgerEvent>, $ReadOnlyArray<LedgerEvent>];

// In ledgerManager:
const [localChanges, remoteChanges] = diffLedger(this._ledger, remoteLedger);
```